beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/path/aws/S3SchemeURI.java

```diff
@@ -17,29 +17,27 @@
 
 import static java.lang.String.format;
 
-import java.net.URI;
-import java.net.URISyntaxException;
-
 import com.amazonaws.services.s3.AmazonS3URI;
 
 import com.expediagroup.beekeeper.core.error.BeekeeperException;
 
 public class S3SchemeURI {
 
+  private static final String S3_SCHEME = "s3://";
+  private static final String S3_SCHEME_REGEX = "^s3(a|n):\\/\\/";
+
   private AmazonS3URI amazonS3URI;
 
-  public S3SchemeURI(String housekeepingPath) throws URISyntaxException {
+  public S3SchemeURI(String housekeepingPath) {
     if (!housekeepingPath.startsWith("s3")) {
       throw new BeekeeperException(format("'%s' is not an S3 path.", housekeepingPath));
     }
-    URI uri = URI.create(housekeepingPath);
-    URI s3Uri = new URI("s3", uri.getHost(), uri.getPath(), null);
-    this.amazonS3URI = new AmazonS3URI(s3Uri);
+    String s3Path = housekeepingPath.replaceFirst(S3_SCHEME_REGEX, S3_SCHEME);
+    this.amazonS3URI = new AmazonS3URI(s3Path);
```

Comment:
And we don't need to pass in the boolean `true` as an additional param like we do in the class below?

Reply:
Nope, this is set automatically in this constructor